eslint.shared.js

```diff
@@ -10,15 +10,22 @@ const {
 } = require('eslint-config-galex/dist/overrides/typescript');
 
 module.exports = {
+  /**
+   * Create linting config for monorepo project
+   * @param {string} cwd - pass `__dirname` so galex can find the project's `package.json`
+   * @param {import('eslint-config-galex/dist/types').Dependencies} dependencies - dependencies object returned by `getDependencies()`
+   * @param {(import('eslint').Linter.ConfigOverride | null)[]} customOverrides
+   * @returns {import('eslint-config-galex/dist/types').TopLevelESLintConfig}
+   */
   createConfig: (cwd, dependencies, customOverrides = []) => {
     return createConfig({
       cwd,
       env: { es6: true }, // https://github.com/ljosberinn/eslint-config-galex/pull/664
-      enableJavaScriptSpecificRulesInTypeScriptProject: true, // to lint `.eslintrc.js` files and the like
```

Comment:
Galex provides two options for fully linting JS files in a TypeScript project:

- using the property above, or
- enabling `checkJs` in `tsconfig.json`

I've decided to enable `checkJs` in this repo, which has the added benefit that we can use `tsc` to type check JS/CJS files (once you include them in `tsconfig.json`).

The `enableJavaScriptSpecificRulesInTypeScriptProject` property is more for JS codebases that are migrating to TypeScript (like daiquiri).